pkg/queue/queue.go

```diff
@@ -109,19 +118,36 @@ func (cq *ClusterQueue) PushIfNotPresent(info *workload.Info) bool {
 	if item != nil {
 		return false
 	}
+
+	log := cq.log.WithValues("workload", workload.Key(info.Obj))
+	ctx := ctrl.LoggerInto(context.Background(), log)
+	// Populate priority from priority class before enqueue.
+	p := workload.GetPriorityFromPriorityClass(ctx, cq.client,
+		info.Obj.Spec.PriorityClassName)
+	info.Priority = &p
+
 	heap.Push(&cq.heap, *info)
 	return true
 }
 
-func (cq *ClusterQueue) PushOrUpdate(w *kueue.QueuedWorkload) {
+func (cq *ClusterQueue) PushOrUpdate(w *kueue.QueuedWorkload) bool {
 	item := cq.heap.items[workload.Key(w)]
 	info := *workload.NewInfo(w)
```

Comment:
why not calculate the value in `NewInfo`?

Reply:
I was initially hoping to calculate in NewInfo. But I find WorkloadInfo is a basic data struct which is invoked in other places like cache. It's hard and unclean if I pass the log/ctx/client to everywhere I want to invoke New.